src/pages/workspaces/workspace/workflows/DataStepContent.js

```diff
@@ -86,69 +94,115 @@ export default class DataStepContent extends Component {
+          [
+            div([
+              h(RadioButton, {
+                text: `Create a new set from selected ${baseEntityType}s`,
+                name: 'choose-set-components',
+                checked: isChooseSetComponents,
+                onChange: () => this.setNewSelectionModel({ type: chooseSetComponents, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            div([
+              h(RadioButton, {
+                text: `Create a new set from all ${baseEntityTypeCount} ${baseEntityType}s`,
+                name: 'choose-set-components',
+                checked: isProcessAllSetComponents,
+                onChange: () => this.setNewSelectionModel({ type: processAllSetComponents, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            hasEntityType && div([
+              h(RadioButton, {
+                text: 'Choose existing sets',
+                name: 'choose-set-components',
+                checked: isChooseSets,
+                onChange: () => this.setNewSelectionModel({ type: chooseSets, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ])
+          ] : [
+            div([
+              h(RadioButton, {
+                text: `Process all ${rootEntityTypeCount} rows`,
+                name: 'process-rows',
+                checked: isProcessAll,
+                onChange: () => this.setNewSelectionModel({ type: processAll, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            div([
+              h(RadioButton, {
+                text: 'Choose specific rows to process',
+                name: 'process-rows',
+                checked: isChooseRows,
+                onChange: () => this.setNewSelectionModel({ type: chooseRows, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ]),
+            hasSet && div([
+              h(RadioButton, {
+                text: 'Choose existing sets',
+                name: 'process-rows',
+                checked: isProcessMergedSet,
+                onChange: () => this.setNewSelectionModel({ type: processMergedSet, selectedEntities: {} }),
+                labelStyle: { marginLeft: '0.75rem' }
+              })
+            ])
           ]),
-          hasSet && div([
-            h(RadioButton, {
-              text: 'Choose existing sets',
-              name: 'process-rows',
-              checked: isProcessMergedSet,
-              onChange: () => this.setNewSelectionModel({ type: processMergedSet, selectedEntities: {} }),
-              labelStyle: { marginLeft: '0.75rem' }
-            })
-          ])
-        ]),
-        !isProcessAll && div({
+        (!isProcessAll && !isProcessAllSetComponents) && div({
```

Comment:
We only really use parens if we're mixing `&&` and `||`
```suggestion
        !isProcessAll && !isProcessAllSetComponents && div({
```